### PR TITLE
Don't issue completion in JSX text

### LINKED ISSUE #5096: Completion should not appear at positions of JSX text
Courtesy of @SaschaNaz at https://github.com/Microsoft/TypeScript/issues/5088#issuecomment-145278660

autocompletioninjsxelement

For the "HiHi" part, no completion should be there.

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #5096

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -3311,11 +3311,24 @@ namespace ts {
...
 
+            function isInJsxText(contextToken: Node): boolean {
+                if (contextToken.kind === SyntaxKind.JsxText) {
+                    return true;
+                }
+
+                return contextToken.kind === SyntaxKind.GreaterThanToken &&
+                    contextToken.parent &&
+                    (contextToken.parent.kind === SyntaxKind.JsxOpeningElement ||
+                        contextToken.parent.kind === SyntaxKind.JsxSelfClosingElement ||
```
Comment (mhegazy) [inline src/services/services.ts]: so what if you are on the next line:
```ts
var x = <div />
/*here*/
```

Comment (RyanCavanaugh) [inline src/services/services.ts]: Good point. Revising.

**Comment by mhegazy** [comment]
:+1:

File: src/services/services.ts [hunk 2]
```diff
@@ -3311,11 +3311,29 @@ namespace ts {
...
+                    return true;
+                }
+
+                if (contextToken.kind === SyntaxKind.GreaterThanToken && contextToken.parent) {
+                    if (contextToken.parent.kind === SyntaxKind.JsxOpeningElement) {
+                        return true;
+                    }
+
+                    if (contextToken.parent.kind === SyntaxKind.JsxClosingElement || contextToken.parent.kind === SyntaxKind.JsxSelfClosingElement) {
+                        return contextToken.parent.parent && contextToken.parent.parent.kind === SyntaxKind.JsxElement;
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: Will we always have this level of nesting with certainty?

Comment (RyanCavanaugh) [inline src/services/services.ts]: The nesting structure of closing elements relative to their Element parents is fixed, if that's the question

**Comment by RyanCavanaugh** [comment]
Argh, pushed my fix for #5088 into this PR accidently. @mhegazy can you review the combined changes?

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**